### ISSUE TITLE
IgnoredReturnValue false positives

### ISSUE THREAD

**Issue by yogurtearl** [detekt/detekt#3043]
## Expected Behavior

Using RxJava 2.2.19

```kotlin
Observable.just(...)
            .take(1)
            .map {...
            }
           .kotlinExtensionMethod()
          .subscribe({... }, {...})
```

Should be no `IgnoredReturnValue` detekted becase we use the results of `take` and `map`

## Observed Behavior
This triggered `IgnoredReturnValue` on `take` and `map`

## Context

We have to disable this rule as a workaround.

## Your Environment

* Version of detekt used: 1.12.0
* Version of Gradle used (if applicable): 6.6.1 (with Android)
* Operating System and version: macOS 10.15

**Comment by vladimirfx** [comment]
Same issue with chained method call:

        assertThat(debt.devicesDebts)
            .extracting(Function<DevicesDebts, Instant?> { it.sentDate })
            .containsOnlyNulls()


Here `extracting(..)` is marked with `@CheckReturnValue` but `containsOnlyNulls()` is not. So value returned by `extracting(..)` is consumed.
